.buildkite/pipeline.yml

```diff
@@ -1,80 +1,233 @@
+      - komo ctx staging
+      - helm status -n $$NAMESPACE $$CHART_NAME && helm uninstall $$CHART_NAME -n $$NAMESPACE && sleep 10
+      - BRANCH=$(git rev-parse --short HEAD)
+      - git checkout master
+      - echo "installing old helm chart on staging"
+      - helm install $$CHART_NAME charts/komodor-agent --set clusterName=$$CHART_NAME --set imagePullSecret=docker-cfg-komodorio --namespace=$$NAMESPACE --create-namespace --set communications.serverHost=https://staging.app.komodor.com --set communications.tasksServerHost=https://staging.app.komodor.com --set communications.wsHost=wss://staging.app.komodor.com --set capabilities.telemetry.enable=false --set communications.telemetryServerHost=https://staging.telemetry.komodor.com --set apiKey=$$KOMODOR_AGENT_STAGING_API_KEY --set allowedResources.secret=true --set capabilities.event.redact="{.*KEY.*,.*key.*,.*BUGSNAG.*}"
+      - git checkout $$BRANCH
+      - echo "upgrade to new helm chart on staging"
+      - helm get values $$CHART_NAME -n $$NAMESPACE > old-values.yaml && helm upgrade --install $$CHART_NAME charts/komodor-agent -n $$NAMESPACE -f old-values.yaml
+      - helm uninstall $$CHART_NAME -n $$NAMESPACE
+      - kubectl delete namespace $$NAMESPACE
     agents:
       builder: "dind"
     plugins:
       - zacharymctague/aws-ssm#v1.0.0:
           parameters:
-            STAGING_API_KEY: /komodor/staging/kubernetes/agent-api-key
-    label: "test helm update install"
+            KOMODOR_AGENT_STAGING_API_KEY: /komodor/staging/komodor-agent-chart/api-key
+
     if: build.message !~ /feat\(OSS.+\):/i && build.branch != "master"
   - wait
 
-  - block: "Bump Versions"
+  - block: ":infinity: Bump Versions"
     if: build.message !~ /feat\(OSS.+\):/i && build.branch == "master"
   - commands:
       - "git config user.email buildkite@users.noreply.github.com"
       - "git config user.name buildkite"
       - git fetch --tags
       - git checkout master
-      - APP_VERSION=$(buildkite-agent meta-data get "version" --job ${PARENT_JOB_ID}) || APP_VERSION=$(cat charts/k8s-watcher/Chart.yaml | grep 'appVersion:' | awk '{print \$2}')
+      - git checkout -b bump-versions-test
+      - CHART=k8s-watcher
+      - APP_VERSION=$(buildkite-agent meta-data get "version" --job ${PARENT_JOB_ID}) || APP_VERSION=$(cat charts/$$CHART/Chart.yaml | grep 'appVersion:' | awk '{print \$2}')
+      - echo "Updating app version to $$APP_VERSION"
+      - 'sed -i -e "s/appVersion.*/appVersion: $$APP_VERSION/g" charts/$$CHART/Chart.yaml'
+      - 'buildkite-agent meta-data set "agent-version" "$$APP_VERSION"'
+      - CURRENT_VERSION=$(cat charts/$$CHART/Chart.yaml | grep 'version:' | awk '{print \$2}')
+      - NEW_VERSION=$(echo $$CURRENT_VERSION | awk -F. '{\$NF = \$NF + 1;} 1' | sed 's/ /./g')
+      - echo "Updating chart '$$CHART' version from $$CURRENT_VERSION to $$NEW_VERSION"
+      - 'sed -i -e "s/$$CURRENT_VERSION/$$NEW_VERSION/g" charts/$$CHART/Chart.yaml'
```

Comment:
we might want to organize these in bash/python/makefile scripts, it it becoming very very hard to read especially in CRs.

Reply:
@nirbenator, do we want to invest more time and change? 
We discussed it a few days ago...  the pipeline seems very bad and not user-friendly.

Reply:
Extracted most of the commands to scripts